internal/version/version.go

```diff
@@ -79,3 +80,22 @@ func Get() BuildInfo {
 	}
 	return v
 }
+
+func init() {
+	if gitCommit != "" || gitTreeState != "" {
```

Comment:
Can we make a seperate PR for chaning the git state please? The Kubernetes changes look good to merge for me.

Here, I think we should take `gitCommit` from `vcs.revision`. But to keep the seperate changes independant, think worth creating separate PRs.